src/resolve/resolved-link.c

```diff
@@ -390,6 +390,9 @@ void link_set_dns_over_tls_mode(Link *l, DnsOverTlsMode mode) {
 #endif
 
         l->dns_over_tls_mode = mode;
+
+        if (l->unicast_scope)
```

Comment:
No need for the if check here I think, `dns_scope_free()` does its own check for `NULL` internally